### PR TITLE
Add mongo auth

### PR DESCRIPTION
**Pull request by Lomonosow**
#13  Is work for me, but use mongo admin user. I`d tried to find another way but unsuccessfully)

### CODE REVIEW

**Comment by warrenvw** [comment]
@Lomonosow Thank you for your contribution. I'll be reviewing it shortly.

**Comment by Lomonosow** [comment]
I found some mistake in configuration, mongo want key at least 6 symbol length. I have fix it in my commit.

**Comment by armab** [comment]
@warrenvw Can you please drive this review, think about corner cases and check the PR?

From my point of view it would be nice to add a condition and set mongo username/password in `st2.conf` only if `mongodb-ha.auth.enabled = true`

File: templates/configmaps_st2-conf.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ data:
     url = amqp://{{ required "rabbitmq-ha.rabbitmqUsername is required!" (index .Values "rabbitmq-ha" "rabbitmqUsername") }}:{{ required "rabbitmq-ha.rabbitmqPassword is required!" (index .Values "rabbitmq-ha" "rabbitmqPassword") }}@{{ .Release.Name }}-rabbitmq-ha-discovery:5672
     [database]
     # TODO: ReplicaSet connection string needs templating based on number of MongoDB nodes
-    host = mongodb://{{ .Release.Name }}-mongodb-ha/?replicaSet=rs0
+    host = mongodb://{{ required "mongodb-ha.auth.adminUser is required!" (index .Values "mongodb-ha" "auth" "adminUser") }}:{{ required "mongodb-ha.auth.adminPassword is required!" (index .Values "mongodb-ha" "auth" "adminPassword") }}@{{ .Release.Name }}-mongodb-ha/?authSource=admin&replicaSet=rs0
```
Comment (warrenvw) [inline templates/configmaps_st2-conf.yaml]: Per @armab, add a condition and set mongo username/password in `st2.conf` only if `mongodb-ha.auth.enabled = true`.

File: values.yaml [hunk 2]
```diff
@@ -367,6 +367,11 @@ mongodb-ha:
   image:
     # StackStorm currently supports maximum MongoDB v3.4
     tag: 3.4
+  auth:
+    enabled: true
+    adminUser: "admin"
+    adminPassword: "admin"
+    key: "keykeykey"
```
Comment (warrenvw) [inline values.yaml]: I would suggest adding a note that the default `key` should be changed, similar to the note above `secrets`.

`NB! It's highly recommended to change ALL defaults!`

Comment (Lomonosow) [reply values.yaml]: Added

File: templates/configmaps_st2-conf.yaml (hunk 1 above)
Comment (Lomonosow) [reply templates/configmaps_st2-conf.yaml]: Done

**Comment by warrenvw** [comment]
I'd appreciate if you could update the Changelog and bump the version in `Charts.yaml`. The Changelog should probably indicate that auth is now enabled by default.

Please note that I have not been able to successfully transition mongodb from auth disabled to auth enabled. Have you tried this? I'm attempting this particular scenario because auth is likely disabled on most installations... and now by default it will be enabled.

For example, in the `st2actionrunner` pod, I see:

```
OperationFailure: Authentication failed.
```

**Comment by Lomonosow** [comment]
> For example, in the st2actionrunner pod, I see:
> 
> OperationFailure: Authentication failed.
I have reproduce it when I`d change mongodb admin password and tried to upgrade helm release. Mongodb chart does not recreate user if directory for db already exists (in mongodb persistent storage enabled by default and if you delete release by helm, persistent storage will not be deleted).

**Pull request merged**